broker/data-access-layer/iaas/src/index.js

```diff
@@ -18,9 +19,13 @@ const getCloudClient = function (settings) {
     case 'gcp':
       return new GcpClient(settings);
     case 'aws':
+      return new AwsClient(settings);
     case 'openstack':
     case 'os':
-      return new CloudProviderClient(settings);
+      if(process.env.NODE_ENV !== 'production') {
+        return new CloudProviderClient(settings);
+      }
+      return undefined;
```

Comment:
```suggestion
      return new BaseCloudClient(settings);
```
I think, if we return `undefined`, the cloudProvider will be set to undefined and fail in unexpected areas. May be we can return an instance of BaseCloudClient?

Reply:
Yah, we can do that, though that line would never be executed.

Reply:
@anujbansal16 I think it makes sense to have an instance of BaseCloudClient here if it doesn't come in way of UT execution.. Also can you please check if this can also be done in above L4. IMO, we should avoid adding explicit undefined/null values in codebase.

Reply:
@anujbansal16, Since we are reading settings.name from config file, I think there might be a chance of this case getting hit in production if someone configures `openstack` by mistake. This is very unlikely though. But since config files are easy to change may be better to be safe.

Reply:
oh ok @jayeshjk  and @anoopjb . I will modify it to bring those changes.